src/_pytest/assertion/util.py

```diff
@@ -147,7 +147,9 @@ def assertrepr_compare(config, op, left, right):
     explanation = None
     try:
         if op == "==":
-            if istext(left) and istext(right):
+            if isinstance(left, bytes) and isinstance(right, bytes):
+                explanation = _diff_text(left, right, verbose)
```

Comment:
I don't think we should be treating binary data as text

Reply:
That's what it does on py2 and from all the comments there it looks to just be
an oversight with py3.

So if not doing it for py3 it should be done with unicode strings on py2
probably then also only?

Current:

```
b'\\x00\\x00\\x00' == b'\\x00\\x01\\x02'
At index 1 diff: 0 != 1
Full diff:
- b'\\x00\\x00\\x00'
?          ^   ^
+ b'\\x00\\x01\\x02'
?          ^   ^
```

This:

```
b'\\x00\\x00\\x00' == b'\\x00\\x01\\x02'
- \\x00\\x00\\x00
?        ^   ^
+ \\x00\\x01\\x02
?        ^   ^
```

So the current version is actually more useful.

Reply:
The second version loses the `b'` which is arguably worse

Reply:
That would be easy to not do though - I've done it to match what the function is intended to do (with regard to the py2 behavior).

Reply:
Let's go with your suggestion then?

> we should maybe just omit the "contains more items" messaging for bytes objects?

Reply:
And should the py2 behavior be kept then?

Reply:
Yeah, sounds good. We can keep the py2 behavior as strings and bytes are more interchangeable there

Reply:
Ok, done.